### DISCUSSION TITLE
How to screenshot the designated area of the figure?

### DISCUSSION THREAD

**Discussion by mellow753** [pyvista/pyvista#2298] [Q&A]
Hi, I can output a screenshot with the whole field now, shown here:
![p](https://user-images.githubusercontent.com/57876661/156785430-3679131d-20a0-4fe7-bec3-078e2e72b93b.png)

But I'd like to output a screenshot with the specified area, like:
![2](https://user-images.githubusercontent.com/57876661/156785642-ef25b6a7-78a5-42c4-a443-29b9fb20d511.png)

My code is

```py
import pyvista as pv

p = pv.read('p_zNormal_0.2.vtk')

mesh = pv.PolyData(p)
print(mesh)
print(f"All arrays: {mesh.array_names}")

plotter = pv.Plotter(off_screen=True)
plotter.add_mesh(mesh, scalars='p', cmap="jet")
plotter.view_xy()
plotter.screenshot('p.png', transparent_background=True)
```

Thanks for any suggestion!

**Comment by banesullivan** [comment]
You'll want to get the camera's position and set that before saving a screenshot.

Render your scene interactively, save the camera position, then re-use that camera position

```py
import pyvista as pv

p = pv.read('p_zNormal_0.2.vtk')

mesh = pv.PolyData(p)
print(mesh)
print(f"All arrays: {mesh.array_names}")

# On screen with user interaction
plotter = pv.Plotter()
plotter.add_mesh(mesh, scalars='p', cmap="jet")
plotter.view_xy()
plotter.show()
# Move camera to desired location
...
cpos = plotter.camera_position

# Now take screenshots of the scene with that camera position
plotter = pv.Plotter(off_screen=True)
plotter.add_mesh(mesh, scalars='p', cmap="jet")
plotter.camera_position = cpos
plotter.screenshot('p.png', transparent_background=True)
```

**Reply by mellow753** [reply]
Thanks! I can get the zoom in screenshot by resetting the camera position now. And I have one quick question about how to reset the dimension in y-direction. 

The original camera position is:
[(20.0, 0.0, 172.79006470440083),
 (20.0, 0.0, 0.0),
 (0.0, 1.0, 0.0)]
and the corresponding pressure field is 
![1](https://user-images.githubusercontent.com/57876661/156868644-d5a03846-923e-417e-b7f7-e639570297c4.png)

The modified camera position is:
[(5.0, 0.0, 30.0),
 (5.0, 0.0, 0.0),
 (0.0, 1.0, 0.0)]
and the corresponding pressure field is 
![p_0 2](https://user-images.githubusercontent.com/57876661/156868667-8df45e08-78d0-44eb-91ef-d97a93cfec35.png)

How can I make the y-dimension narrower without changing the x-dimension, by window_size? And how can I move the legend bar to fixed position, like the northwest. Thanks!
